File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    q = deque(enumerate(priorities))
    
    order = 0
    while q:
        idx, priority = q.popleft()
        
        if q and priority < max(p for _, p in q):
            q.append((idx, priority))
        else:
            order += 1
            if idx == location:
                return order
                    
    

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        ArrayDeque<int[]> q = new ArrayDeque<>();
        
        for(int i=0; i<priorities.length; i++){
            q.add(new int[]{i, priorities[i]});
        }
        
        int order = 0;
        
        while(!q.isEmpty()){
            int[] cur = q.poll();
            
            boolean higher = false;
            
            for(int[] next : q){
                if(next[1] > cur[1]){
                    higher = true;
                    break;
                }
            }
            
            if(higher){
                q.add(cur);
            }
            else{
                order++;
                
                if(cur[0] == location){
                    return order;
                }
            }
        }
        return -1;
    }
}